external/CMakeLists.txt

```diff
@@ -59,4 +58,40 @@ if (${ENABLE_SPIRV_CODEGEN})
       message(FATAL_ERROR "GoogleTest was not found - required for SPIR-V codegen")
     endif()
   endif()
-endif()
\ No newline at end of file
+
+  # We need Effcee for SPIR-V codegen testing.
+  if (${SPIRV_BUILD_TESTS})
+    set(DXC_RE2_DIR "${DXC_EXTERNAL_ROOT_DIR}/re2" CACHE STRING "Location of RE2 source")
+    set(DXC_EFFCEE_DIR "${DXC_EXTERNAL_ROOT_DIR}/effcee" CACHE STRING "Location of Effcee source")
+
+    if (IS_DIRECTORY ${DXC_RE2_DIR})
+      # Silence some level-3 and level-4 warnings from RE2.
+      add_compile_options(/wd4201 /wd4815 /EHs)
+      # Don't build/run re2's tests.
+      set(RE2_BUILD_TESTING OFF CACHE BOOL "Skip RE2 tests")
+      add_subdirectory(${DXC_RE2_DIR})
+    endif()
+
+    if (IS_DIRECTORY ${DXC_EFFCEE_DIR})
+      # Configure effcee
+      set(EFFCEE_RE2_DIR "${DXC_RE2_DIR}" CACHE STRING "Location of RE2 for effcee")
+      set(EFFCEE_GOOGLETEST_DIR "${DXC_GTEST_DIR}" CACHE STRING "Location of googletest for effcee")
```

Comment:
I don't think `EFFCEE_GOOGLETEST_DIR` is necessary since we are not building effcee tests any more?